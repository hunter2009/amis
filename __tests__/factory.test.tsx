import {
    registerRenderer,
    unRegisterRenderer,
    RendererProps,
    render as amisRender
} from '../src/factory';
import React = require('react');
import {render, fireEvent, cleanup} from 'react-testing-library';

test('factory unregistered Renderer', () => {
    const {
        container,
    } = render(amisRender({
        type: 'my-renderer',
        a: 23
    }));

    expect(container).toMatchSnapshot(); // not found
});

test('factory:registerRenderer', () => {
    interface MyProps extends RendererProps {
        a: number;
    };

    class MyComponent extends React.Component<MyProps> {
        render() {
            return (<div>This is Custom Renderer, a is {this.props.a}</div>);
        }
    }

    const renderer = registerRenderer({
        component: MyComponent,
        test: /\bmy-renderer$/
    });

    const {
        container
    } = render(amisRender({
        type: 'my-renderer',
        a: 23
    }))

    expect(container).toMatchSnapshot();
    unRegisterRenderer(renderer);
});