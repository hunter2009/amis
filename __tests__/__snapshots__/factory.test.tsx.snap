// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factory unregistered Renderer 1`] = `
<div>
  <div
    class="Alert Alert--danger"
  >
    <p>
      Error: 找不到对应的渲染器
    </p>
    <p>
      Path: 
      my-renderer
    </p>
    <pre>
      <code>
        {
  "type": "my-renderer",
  "a": 23
}
      </code>
    </pre>
  </div>
</div>
`;

exports[`factory:registerRenderer 1`] = `
<div>
  <div>
    This is Custom Renderer, a is 
    23
  </div>
</div>
`;
